app/src/services/tag.js

```diff
@@ -284,14 +284,15 @@ const service = {
       })
       .modify('filterId', params.versionIds)
       // filter by objects that user(s) has READ permission at object or bucket-level
-      // TODO: consider instead doing `.whereIn('version.objectId', <objects with permission>)`
       .modify((query) => {
         if (params.userId) {
           query
-            .allowGraph('object.[objectPermission, bucketPermission]')
-            .withGraphJoined('object.[objectPermission, bucketPermission]')
-            .modifyGraph('object', query => { query.modify('hasPermission', params.userId, 'READ'); })
-            .whereNotNull('object.id');
+            .allowGraph('object')
+            .withGraphJoined('object')
+            .modifyGraph('object', query => {
+              query
+                .modify('hasPermission', params.userId, 'READ');
```

Comment:
Optional style: could make this one line as we're not really chaining other things for this query.